datashare-app/src/main/java/org/icij/datashare/utils/DocumentVerifier.java

```diff
@@ -60,4 +54,18 @@ private long getEmbeddedDocumentDownloadMaxSizeBytes() {
         String maxSize = propertiesProvider.get(EMBEDDED_DOCUMENT_DOWNLOAD_MAX_SIZE_OPT).orElse(DEFAULT_MAX_SIZE);
         return HumanReadableSize.parse(maxSize);
     }
+
+    /**
+     * Retrieves the root document for an embedded document even if the root is a duplicate.
+     *
+     * @return The max size in bytes.
+     */
+    private Document getRootDocument(Document document) {
+        Entity entity = indexer.get(document.getProjectId(), document.getRootDocument());
+        if (entity.getClass().equals(Duplicate.class)) {
+            return indexer.get(document.getProjectId(), ((Duplicate) entity).documentId);
+        } else {
+            return (Document) entity;
+        }
```

Comment:
Thanks for the change! We also generally avoid to use else after return (for instance in pylint it's the rule called [`no-else-return`](https://pheanex.github.io/pylint/#R1705)). So this would be more elegant:


```suggestion
        }
        return (Document) entity;
```